src/pykeen/nn/modules.py

```diff
@@ -0,0 +1,1121 @@
+            The head representations.
+        :param r: shape: (batch_size, d_r)
+            The relation representations.
+        :param t: shape: (batch_size, d_e)
+            The tail representations.
+
+        :return: shape: (batch_size, 1)
+            The scores.
+        """
+        return self._score(h=h, r=r, t=t)[:, 0, 0, 0, None]
+
+    def score_h(
+        self,
+        all_entities: HeadRepresentation,
+        r: RelationRepresentation,
+        t: TailRepresentation,
+        slice_size: Optional[int] = None,
+    ) -> torch.FloatTensor:
+        """Score all head entities.
+
+        :param all_entities: shape: (num_entities, d_e)
+            The head representations.
+        :param r: shape: (batch_size, d_r)
+            The relation representations.
+        :param t: shape: (batch_size, d_e)
+            The tail representations.
+        :param slice_size:
+            The slice size.
+
+        :return: shape: (batch_size, num_entities)
+            The scores.
+        """
+        return self._score(h=all_entities, r=r, t=t, slice_dim="h", slice_size=slice_size)[:, :, 0, 0]
+
+    def score_r(
+        self,
+        h: HeadRepresentation,
+        all_relations: RelationRepresentation,
+        t: TailRepresentation,
+        slice_size: Optional[int] = None,
+    ) -> torch.FloatTensor:
+        """Score all relations.
+
+        :param h: shape: (batch_size, d_e)
+            The head representations.
+        :param all_relations: shape: (num_relations, d_r)
+            The relation representations.
+        :param t: shape: (batch_size, d_e)
+            The tail representations.
+        :param slice_size:
+            The slice size.
+
+        :return: shape: (batch_size, num_entities)
+            The scores.
+        """
+        return self._score(h=h, r=all_relations, t=t, slice_dim="r", slice_size=slice_size)[:, 0, :, 0]
+
+    def score_t(
+        self,
+        h: HeadRepresentation,
+        r: RelationRepresentation,
+        all_entities: TailRepresentation,
+        slice_size: Optional[int] = None,
+    ) -> torch.FloatTensor:
+        """Score all tail entities.
+
+        :param h: shape: (batch_size, d_e)
+            The head representations.
+        :param r: shape: (batch_size, d_r)
+            The relation representations.
+        :param all_entities: shape: (num_entities, d_e)
+            The tail representations.
+        :param slice_size:
+            The slice size.
+
+        :return: shape: (batch_size, num_entities)
+            The scores.
+        """
+        return self._score(h=h, r=r, t=all_entities, slice_dim="t", slice_size=slice_size)[:, 0, 0, :]
+
+    def reset_parameters(self):
+        """Reset parameters the interaction function may have."""
+        for mod in self.modules():
+            if mod is self:
+                continue
+            if hasattr(mod, 'reset_parameters'):
+                mod.reset_parameters()
+
+
+class LiteralInteraction(
```

Comment:
this doesn't get tested in this PR since it's not used in a model yet. maybe there's a way to test it without moving code that does the next step?